src/NLog.Targets.Syslog/MessageSend/Tcp.cs

```diff
@@ -138,8 +142,7 @@ private Task WriteAsync(int offset, ByteArray data, CancellationToken token)
             var isLastWrite = toBeWrittenTotal <= dataChunkSize;
             var count = isLastWrite ? toBeWrittenTotal : dataChunkSize;
 
-            return Task.Factory
-                .SafeFromAsync(stream.BeginWrite, stream.EndWrite, (byte[])data, offset, count, null)
+            return stream.WriteAsync((byte[])data, offset, count)
```

Comment:
WriteAsync internally relies on Task.Factory.FromAsync which does not create a faulted Task when a Socket exceptions occurs: that's why SafeFromAsync should be kept